src/catalog/catalog_entry/duck_table_entry.cpp

```diff
@@ -754,6 +726,51 @@ unique_ptr<CatalogEntry> DuckTableEntry::DropForeignKeyConstraint(ClientContext
 	return make_uniq<DuckTableEntry>(catalog, schema, *bound_create_info, storage);
 }
 
+unique_ptr<CatalogEntry> DuckTableEntry::AddConstraint(ClientContext &context, AddConstraintInfo &info) {
+	auto create_info = make_uniq<CreateTableInfo>(schema, name);
+	create_info->comment = comment;
+
+	// Copy all columns and constraint to the modified table
+	create_info->columns = columns.Copy();
+	for (auto &constraint : constraints) {
+		create_info->constraints.push_back(constraint->Copy());
+	}
+
+	switch (info.constraint->type) {
+	case ConstraintType::UNIQUE: {
+		auto unique = reinterpret_cast<UniqueConstraint *>(info.constraint.get());
```

Comment:
We have `info.constraint->Cast<UniqueConstraint>()` for this purpose

Reply:
I removed all appearances of this for `*Constraint` classes from the entire codebase.